sources/network/windows/io_service.cpp

```diff
@@ -151,11 +193,16 @@ io_service::process_rd_event(const fd_t& fd, tracked_socket& socket) {
     rd_callback(fd);
 
     std::lock_guard<std::mutex> lock(m_tracked_sockets_mtx);
-    auto it = m_tracked_sockets.find(fd);
+    auto it      = m_tracked_sockets.find(fd);
+    auto& socket = it->second;
 
-    if (it == m_tracked_sockets.end()) { return; }
+    if (it == m_tracked_sockets.end()) {
+      //Make sure we clean up the flags and notify...
+      socket.is_executing_rd_callback = false;
```

Comment:
should not be necessary: it iterator is end, then socket is invalid memory, so we are clearing nothing.
This cleanup should probably be moved somewhere else, when the socket is effectively removed from the track list.